Neos.Fusion/Classes/Core/Cache/RuntimeContentCache.php

```diff
@@ -246,7 +249,7 @@ public function postProcess(array $evaluateContext, $fusionObject, $output)
             }
             $cacheTags = $this->buildCacheTags($evaluateContext['configuration'], $evaluateContext['fusionPath'], $fusionObject);
             $cacheMetadata = array_pop($this->cacheMetadata);
-            $output = $this->contentCache->createDynamicCachedSegment($output, $evaluateContext['fusionPath'], $this->serializeContext($contextVariables), $evaluateContext['cacheIdentifierValues'], $cacheTags, $cacheMetadata['lifetime'], $evaluateContext['cacheDiscriminator']);
+            $output = $this->contentCache->createDynamicCachedSegment($output, $evaluateContext['fusionPath'], $this->serializeContext($contextVariables), $evaluateContext['cacheIdentifierValues'], $cacheTags, $cacheMetadata['lifetime'], (string)$evaluateContext['cacheDiscriminator']);
```

Comment:
This one might be interesting.. Without this change, the functional tests fail because`$evaluateContext['cacheDiscriminator']` is `false` at some point.

Reply:
indeed:

```
1) Neos\Fusion\Tests\Functional\FusionObjects\ContentCacheTest::dynamicSegmentCacheBehavesLikeUncachedIfDiscriminatorIsDisabled
TypeError: md5(): Argument #1 ($string) must be of type string, false given

/Cache/Code/Flow_Object_Classes/Neos_Fusion_Core_Cache_ContentCache.php:161
```

and without this cast we wouldnt support this usecase you wrote a test for ^^

https://github.com/neos/neos-development-collection/blob/6ca2ecebb789ea9ed300f92507f2bcbf25695b77/Neos.Fusion/Tests/Functional/FusionObjects/Fixtures/Fusion/ContentCache.fusion#L576

```
@cache.entryDiscriminator = ${false}
```